server/index.js

```diff
@@ -24,79 +23,93 @@ app.use('/', expressStaticGzip(path.join(__dirname, '../client'), {
 
 app.use(cors());
 
-// ----------------- CRUD API REQUESTS ------------------ //
+// ---------------- CRUD FOR POSTGRESQL --------------- //
 
-app.get('/songdata/:id', async (req, res) => {
-  try {
-    var id = req.params.id;
-    if (id > 100 || id < 0) {
-      throw new Error(`songID: ${id} does not exist`);
-    }
-    var song = await Song.findOne({ songID: id });
-    res.status(200).json({
-      status: "Success",
-      data: song
-    });
-  } catch (error) {
-    console.log(error)
-  }
-});
-
-// app.get('/songsByBandID/:id', async (req, res) => {
-//   try {
-//     var bandID = req.params.id;
-//     if (bandID > 30 || bandID < 0) {
-//       res.end('BAND ID DOES NOT EXIST');
-//     } else {
-//       var findSongsByBand = await songData.findSongsByBand(bandID)
-//       res.send(findSongsByBand);
-//     }
-//   } catch (error) {
-//     console.log(error);
-//   }
-// });
+// CREATE
 
 app.post('/song', async (req, res) => {
   try {
-    var song = await Song.create(req.body);
+    var song = await knex('song_list').insert(req.body);
+    // console.log('CREATE SONG request data: ', req.body);
     res.status(201).json({
       status: 'Success',
-      data: song
+      data: req.body
     });
   } catch (error) {
-    console.log('Error creating new song: ', error);
+    console.log(chalk.red('Error creating new song: ', error));
     res.status(400).json({
       status: 'Failed',
       data: error
     });
   }
 });
 
+// READ
+
+app.get('/songdata/:id', async (req, res) => {
+  try {
+    var id = req.params.id;
+    if (id > 10000000 || id < 0) {
+      res.status(404).json({
+        status: 'Not Found',
+        data: null
+      });
+    } else {
+      var song = await knex('song_list').where({ song_id: id });
+      // console.log({
+      //   song
+      // });
+      res.status(200).json({
+        status: 'Success',
+        data: song
+      });
+    }
+  } catch (error) {
+    console.log(chalk.red('Error retrieving song: ', error));
+    res.status(500).json({
+      status: 'Failed',
+      data: null
+    })
+  }
+});
+
+// UPDATE
+
 app.put('/song/:id', async (req, res) => {
   try {
-    var updatedSong = await songData.updateSong(req.params.id, req.body);
+    var updatedSong = await knex('song_list').where({song_id: req.params.id}).update(req.body);
     res.status(200).json({
       status: 'Success',
-      data: updatedSong
+      data: req.body
     });
   } catch (error) {
-    console.log('Error updating song: ', error);
+    console.log(chalk.red('Error updating song: ', error));
     res.status(400).json({
       status: 'Failed',
       data: error
     });
   }
 });
 
+// DELETE
+
 app.delete('/songdata/:id', async (req, res) => {
+  var id = req.params.id;
   try {
-    var deletedSong = await songData.deleteSong(req.params.id);
-    res.status(202).json({
-      status: 'Deleted requested song!',
-      data: null
-    });
+    if (id < 0) {
```

Comment:
Can you send a negative number as an `id` here? Have you tested?

Reply:
No I haven't. This was just part of the legacy codebase